my-app/src/App.js

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+import logo from "./logo.svg";
+import "./reset.css";
+import "./App.css";
+import Nav from "./components/Nav.js";
+import Post from "./components/Post.js";
+import Posts from "./components/Posts.js";
+import More from "./components/More.js";
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = { posts: [], currentPage: 0 };
+  }
+
+  moreBtnClickHandler = e => {
+    this.setState(currentState => {
+      return {
+        posts: currentState.posts,
+        currentPage: ++currentState.currentPage
+      };
+    });
+  };
+
+  componentDidMount() {
+    this.fetchTopStories();
+  }
+
+  fetchTopStories() {
+    fetch("https://hacker-news.firebaseio.com/v0/topstories.json")
+      .then(response => response.json())
+      .then(data => {
+        data.map(storyId => {
```

Comment:
This is very resource intensive (i.e. one request per story) and could easily break for someone on a slow computer or connection.

At the bare minimum, limit the stories you need to fetch to the ones you are displaying on the current page.